src/Voron/Impl/IPagerLevelTransactionState.cs

```diff
@@ -8,6 +8,8 @@ public interface IPagerLevelTransactionState : IDisposable
     {
         Dictionary<AbstractPager, TransactionState> PagerTransactionState32Bits { get; set; }
         Dictionary<AbstractPager, CryptoTransactionState> CryptoPagerTransactionState { get; set; }
+        long TotalEncryptionBufferSize { get; }
```

Comment:
Can we change its type to `Size`? The usage of `Size` make the usage of it a bit less error prone

Reply:
I have changed it to long so we won't have to allocate size object every time we check the size. This is down this way everywhere else I have looked.

Reply:
The reason we have `Size` is that it makes it more difficult to make a mistake like - oh, that's in bytes, I thought the value is in kilobytes etc. `Size` is a struct so it won't have expensive allocation impact.

Reply:
@arekpalinski on the other hand, we could add 'InBytes' to the end of the property name, achieving the same readability, without impacting potential JIT optimizations, due to lack of ability to optimize structs to registers usage.

Reply:
We use `Size` pretty commonly in many places so I'm not so worried about its impact on JIT but we might ask @redknightlois about opinion.

Adding 'InBytes'  prefix is fine but the usage of `Size` is even better because it has dedicated methods, operators etc and given that we use `Size` in other places too then it's easier to deal with them IMO e.g. like here just using `>` operator:

https://github.com/ravendb/ravendb/blob/843cf60f5341a57838e439265a77f1319494fb81/src/Raven.Server/Documents/Indexes/Index.cs#L3336

What do you think?

Reply:
Although it seems t's more of a style issue because both approaches are used throught the code, , while Size demands calling contructors and function calls even for increasing values (while in the bottom line it relies on an inner propery "_valueInBytes") using just a `long` type that represents value in bytes, when it's the only form of data that we are interested in is the simplest, most readable way to go. 
I think that Size starts to be significant when you are required to switch measurement units, therefore I would prefere a cleaner approach here. 
Generally speaking, I think that we should be careful not to make `Size` the default tool, because it might introduce coding standards that may impact overall performance.

Reply:
Size is a struct, should not impact performance that much. It was introduced to avoid miscalculations in the code in places like that (which we had few times in the past).

Reply:
1. Let's use struct here.
2. @maximburyak The JIT is smart enough to inline such a process and eliminate the costs.

Reply:
Use `Size` whenever possible. The cost of `Size` in allocations is 0 and in parameter passing, it is 2 registers, which is negligible. The positive impact it brings in reliability is so big that it is a no-brainer.

Reply:
@redknightlois I think that only in .net 3.0 structs will be optimized into registers. Anyway, you could look into a little comparison I made, it's actually interesting: https://gist.github.com/maximburyak/23dde736e6d5d4178cbcbbc5b6c7f126 bottom line is that run time of Size.Add is more than an order of magnitude slower than long +=, and new Size is two time slower than the first one

Reply:
@maximburyak There are many things wrong on that microbenchmark setup, we cannot trust those results. For proper setup, use BenchmarkDotNet which will handle most of the nuisances (albeit not all of them) so the probability of getting a proper microbenchmark from the get go are far higher.

Aside from that, there is no reason why you would want to do something like this in a hot loop:
` for (Size k = new Size(0, SizeUnit.Bytes); k < maxSize; k.Add(1, SizeUnit.Bytes));`
You are obviously paying the cost of the constructor, and the `.Add()` which may not even get inlined depending on the context. Inside a hot loop, using the long is OK... but `Size` should be used for exchanging and cold checks like the one at the `Index`.

https://twitter.com/federicolois/status/1117762997072617472
This is one of the cases where uglification is not warranted. 

Struct promotion for 2 longs structs already exist since 1.1, it was limited back then but kinda good, On 3.0 it is just amazing.

Reply:
In my(subjective) eyes, using `+=` is at least as readable and elegant and safe as `.Add` or `new Size(` in that particular case, where we are tracking byte count. 
I think that this discussion is more fundamental than the specific case it is initiated for.

Anyway, for completeness, here is a benchmark.net output:

```
// * Summary *

BenchmarkDotNet=v0.11.5, OS=Windows 10.0.17134.765 (1803/April2018Update/Redstone4)
Intel Core i7-8700K CPU 3.70GHz (Coffee Lake), 1 CPU, 12 logical and 6 physical cores
Frequency=3609377 Hz, Resolution=277.0561 ns, Timer=TSC
  [Host]     : .NET Framework 4.7.2 (CLR 4.0.30319.42000), 32bit LegacyJIT-v4.7.3416.0
  DefaultJob : .NET Framework 4.7.2 (CLR 4.0.30319.42000), 32bit LegacyJIT-v4.7.3416.0


|          Method |      Mean |     Error |    StdDev |
|---------------- |----------:|----------:|----------:|
|   LongIncrement | 0.0110 ns | 0.0016 ns | 0.0014 ns |
|   SizeIncrement | 1.5476 ns | 0.0085 ns | 0.0071 ns |
| SizeConstructor | 3.0374 ns | 0.0058 ns | 0.0054 ns |

// * Warnings *
ZeroMeasurement
  MyBenchmark.LongIncrement: Default -> The method duration is indistinguishable from the empty method duration

// * Hints *
Outliers
  MyBenchmark.LongIncrement: Default -> 1 outlier  was  removed (1.19 ns)
  MyBenchmark.SizeIncrement: Default -> 2 outliers were removed (2.75 ns, 2.77 ns)

// * Legends *
  Mean   : Arithmetic mean of all measurements
  Error  : Half of 99.9% confidence interval
  StdDev : Standard deviation of all measurements
  1 ns   : 1 Nanosecond (0.000000001 sec)

// ***** BenchmarkRunner: End *****
// ** Remained 0 benchmark(s) to run **
Run time: 00:01:13 (73.8 sec), executed benchmarks: 3

Global total time: 00:01:15 (75.17 sec), executed benchmarks: 3
// * Artifacts cleanup *
```

See code here: https://gist.github.com/maximburyak/3f83611c0e1a5dabd2da9943d2b53549

Reply:
Not a big difference, and can be fixed if you aggressively inline the add and the constructor which by default is not applied. Either way, while agree about the `Add()` vs `+=`, the addition of the scale far outweigh the performance impact (in this case).